batcher/aligned-batcher/src/metrics.rs

```diff
@@ -1,67 +1,74 @@
-    BATCHER_STARTED.reset();
-
-    OPEN_CONNECTIONS.set(0);
-
-    RECEIVED_PROOFS.reset();
-
-    SENT_BATCHES.reset();
-
-    REVERTED_BATCHES.reset();
-
-    GAS_PRICE_USED_ON_LATEST_BATCH.set(0);
+use warp::{Filter, Rejection, Reply};
+
+#[derive(Clone, Debug)]
+pub struct BatcherMetrics {
+    pub open_connections: IntGauge,
+    pub received_proofs: IntCounter,
+    pub sent_batches: IntCounter,
+    pub reverted_batches: IntCounter,
+    pub batcher_started: IntCounter,
+    pub gas_price_used_on_latest_batch: IntGauge,
 }
 
-pub fn batcher_started() {
-    thread::sleep(time::Duration::from_secs(10));
-    BATCHER_STARTED.inc();
+impl BatcherMetrics {
+    pub fn start(metrics_port: u16) -> anyhow::Result<Self> {
+        let registry = prometheus::Registry::new();
+
+        let open_connections = register_int_gauge!(opts!("open_connections", "Open Connections"))?;
+        let received_proofs = register_int_counter!(opts!("received_proofs", "Received Proofs"))?;
+        let sent_batches = register_int_counter!(opts!("sent_batches", "Sent Batches"))?;
+        let reverted_batches =
+            register_int_counter!(opts!("reverted_batches", "Reverted Batches"))?;
+        let batcher_started = register_int_counter!(opts!("batcher_started", "Batcher Started"))?;
+        let gas_price_used_on_latest_batch =
+            register_int_gauge!(opts!("gas_price_used_on_latest_batch", "Gas Price"))?;
+
+        registry.register(Box::new(open_connections.clone()))?;
+        registry.register(Box::new(received_proofs.clone()))?;
+        registry.register(Box::new(sent_batches.clone()))?;
+        registry.register(Box::new(reverted_batches.clone()))?;
+        registry.register(Box::new(batcher_started.clone()))?;
+
+        let metrics_route = warp::path!("metrics")
+            .and(warp::any().map(move || registry.clone()))
+            .and_then(BatcherMetrics::metrics_handler);
+
+        tokio::task::spawn(async move {
+            warp::serve(metrics_route)
+                .run(([0, 0, 0, 0], metrics_port))
+                .await;
+        });
+
+        Ok(Self {
+            open_connections,
+            received_proofs,
+            sent_batches,
+            reverted_batches,
+            batcher_started,
+            gas_price_used_on_latest_batch,
+        })
+    }
+
+    pub async fn metrics_handler(registry: prometheus::Registry) -> Result<impl Reply, Rejection> {
+        use prometheus::Encoder;
+        let encoder = prometheus::TextEncoder::new();
+
+        let mut buffer = Vec::new();
+        if let Err(e) = encoder.encode(&registry.gather(), &mut buffer) {
+            eprintln!("could not encode prometheus metrics: {}", e);
+        };
+        let res = match String::from_utf8(buffer.clone()) {
+            Ok(v) => v,
+            Err(e) => {
+                eprintln!("prometheus metrics could not be from_utf8'd: {}", e);
+                String::default()
+            }
+        };
```

Comment:
very pedantic nit :), learned this one from Pablo
```suggestion
    let res = String::from_utf8(buffer.clone())
        .inspect_err(|e| eprintln!("prometheus metrics could not be from_utf8'd: {e}"))
        .unwrap_or_default();
```
```

Reply:
also I think the error message is missing some word

Reply:
Nice! I also fixed the comment [78dfcbe](https://github.com/yetanotherco/aligned_layer/pull/1236/commits/78dfcbe5c08913963b71ed5171312450993bb1e7)